examples/marker.html

```diff
@@ -0,0 +1,236 @@
+  renderer.autoClear = false;
+  canvas = renderer.domElement;
+  document.body.appendChild(canvas);
+  scene = new THREE.Scene();
+
+  // Creating the ARView, which is the object that handles
+  // the rendering of the camera stream behind the three.js
+  // scene
+  arView = new THREE.ARView(vrDisplay, renderer);
+
+  // The ARPerspectiveCamera is very similar to THREE.PerspectiveCamera,
+  // except when using an AR-capable browser, the camera uses
+  // the projection matrix provided from the device, so that the
+  // perspective camera's depth planes and field of view matches
+  // the physical camera on the device.
+  camera = new THREE.ARPerspectiveCamera(vrDisplay, 60, window.innerWidth / window.innerHeight, 0.01, 100);
+
+  // Create a model to be placed in the world using picking
+  model = new THREE.Mesh(new THREE.ConeBufferGeometry(
+    MODEL_SIZE_IN_METERS / 2, MODEL_SIZE_IN_METERS), 
+    new THREE.MeshLambertMaterial( {color: 0x888888 } ));
+  // model = new THREE.Mesh(new THREE.SphereBufferGeometry(
+  //   MODEL_SIZE_IN_METERS / 2, 20, 20), 
+  //   new THREE.MeshLambertMaterial( {color: 0x888888 } ));
+  // Apply a rotation to the model so it faces in the direction of the
+  // normal of the plane when the picking based reorientation is done
+  model.geometry.applyMatrix(
+    new THREE.Matrix4().makeTranslation(0, MODEL_SIZE_IN_METERS / 2, 0));
+  model.geometry.applyMatrix(
+    new THREE.Matrix4().makeRotationX(THREE.Math.degToRad(90)));
+  model.position.set(0, 0, -1);
+  scene.add(model);
+  // Add some lighting
+  var directionalLight = new THREE.DirectionalLight(0xffffff, 0.5);
+  directionalLight.position.set(0, 1, 0);
+  scene.add(directionalLight);
+  
+  // VRControls is a utility from three.js that applies the device's
+  // orientation/position to the perspective camera, keeping our
+  // real world and virtual world in sync.
+  vrControls = new THREE.VRControls(camera);
+
+  // Correctly handle window resize events
+  window.addEventListener( 'resize', onWindowResize, false );
+
+  update();
+}
+
+/**
+ * On window resize, update the perspective camera's aspect ratio,
+ * and call `updateProjectionMatrix` so that we can get the latest
+ * projection matrix provided from the device
+ */
+function onWindowResize () {
+  camera.aspect = window.innerWidth / window.innerHeight;
+  camera.updateProjectionMatrix();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+}
+
+function update() {
+    // Render the device's camera stream on screen first of all.
+  // It allows to get the right pose synchronized with the right frame.
+  arView.render();
+
+  // Update our camera projection matrix in the event that
+  // the near or far planes have updated
+  camera.updateProjectionMatrix();
+
+  // Update our perspective camera's positioning
+  vrControls.update();
+
+  // Render our three.js virtual scene
+  renderer.clearDepth();
+  renderer.render(scene, camera);
+
+  if (vrDisplay) {
```

Comment:
I don't think this conditional is necessary, we never enter the render loop if there's no valid AR device

Reply:
done